### PR TITLE
Work for IOS-223

### PR DESCRIPTION
**Pull request by SilviuOdobescu**

### CODE REVIEW

File: AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+
+#pragma mark - Private methods
+- (AlfrescoURLType) parseURLForAction: (NSString *)URLString
+{
+    //Removing the scheme from the url string
+    NSString *urlWithoutScheme = [URLString stringByReplacingOccurrencesOfString:kHandlerPrefix withString:@""];
+    
+    if([urlWithoutScheme hasPrefix:kDocumentPath])
+    {
+        return AlfrescoURLTypeDocument;
```
Comment (tmughal) [inline AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m]: We decided not to use inline returns. Instead do something like this:

```
NSString *urlType = AlfrescoURLTypeNone;

if ([some condition])
{
    urlType = AlfrescoURLTypeDocument;
}

return urlType;
```

File: AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+    }
+    
+    if([[UniversalDevice rootMasterViewController] isKindOfClass:[MainMenuViewController class]])
+    {
+        MainMenuViewController *mainMenu = (MainMenuViewController *)[UniversalDevice rootMasterViewController];
+        [mainMenu cleanSelection];
+    }
+}
+
+- (SwitchViewController *)getPresentingViewController
```
Comment (tmughal) [inline AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m]: "get" is generally not objective-c convention.

File: AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+{
+    if([[UniversalDevice rootDetailViewController] isKindOfClass:[DetailSplitViewController class]])
```
Comment (tmughal) [inline AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m]: can you not just use IS_IPAD here?

File: AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+    {
+        //this is the iPad version
+        DetailSplitViewController *splitViewController = (DetailSplitViewController *)[UniversalDevice rootDetailViewController];
+        if([splitViewController.masterViewController isKindOfClass:[SwitchViewController class]])
+        {
+            SwitchViewController *switchController = (SwitchViewController *)splitViewController.masterViewController;
+            return switchController;
```
Comment (tmughal) [inline AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m]: Inline returns again

File: AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+    {
+        //this is the iPhone version
+        SwitchViewController *switchController = (SwitchViewController *)[UniversalDevice rootDetailViewController];
+        return switchController;
+    }
+    
+    return nil;
+}
+
+- (BOOL) handleURL:(NSURL *)url session:(id<AlfrescoSession>)session
```
Comment (tmughal) [inline AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m]: Minor: space

File: AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+{
+    BOOL handled = NO;
+    AlfrescoURLType actionType = [self parseURLForAction: url.absoluteString];
```
Comment (tmughal) [inline AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m]: Minor: space

File: AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+    self.viewControllerToPresent = nil;
+    switch (actionType)
+    {
+        case AlfrescoURLTypeNone:
+        {
+            handled = YES;
+        }
+            break;
```
Comment (tmughal) [inline AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m]: Minor: indention

File: AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+            UILabel *newLabel = [[UILabel alloc] initWithFrame:CGRectMake(0, 0, 320, 100)];
+            newLabel.text = username;
+            [controller.view addSubview: newLabel];
+            self.viewControllerToPresent = controller;
+            handled = YES;
+        }
+            break;
+    }
+    
+    if(self.viewControllerToPresent)
```
Comment (tmughal) [inline AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m]: Minor: spacing

File: AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+@interface AlfrescoURLHandler() < AKUserAccountListViewControllerDelegate >
+
+@property (nonatomic, strong) UIViewController *viewControllerToPresent;
+@property (nonatomic, strong) NSURL *urlReceived;
+
+@end
+
+@implementation AlfrescoURLHandler
+
+static NSString * const kHandlerPrefix = @"alfresco://";
```
Comment (tmughal) [inline AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m]: Group all constant strings together at one place in the .m file please.

File: AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m [hunk 10]
```diff
@@ -0,0 +1,281 @@
...
+            handled = [self handleURL:url session:session];
+        }
+    }
+
+    
+    return handled;
+}
+
+#pragma mark - Private methods
+- (AlfrescoURLType) parseURLForAction: (NSString *)URLString
```
Comment (tmughal) [inline AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m]: Minor: spacing

File: AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m (hunk 3 above)
Comment (SilviuOdobescu) [inline AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m]: As it stands today yes I think that IS_IPAD could be used but I think this condition would have to be changed once support for iPhone6+ is added.

Comment (tmughal) [inline AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m]: Fair enough, but wouldn't this also need to change as support for the iPhone 6+ would use size classes?

**Pull request merged**